lib/spack/spack/cmd/buildcache.py

```diff
@@ -60,8 +60,10 @@ def setup_parser(subparser):
                                             "building package(s)")
     create.add_argument('-y', '--spec-yaml', default=None,
                         help='Create buildcache entry for spec from yaml file')
-    create.add_argument('--no-deps', action='store_true', default='false',
-                        help='Create buildcache entry wo/ dependencies')
+    create.add_argument('--only', default='package,dependencies',
+                        dest='things_to_install',
+                        choices=['package', 'dependencies'],
+                        help="add package and/or its dependencies")
```

Comment:
Presumably the default is to add both?

From the `spack install` help string: `select the mode of installation. the default is to install the package along with all its dependencies. alternatively one can decide to install only the package or only the dependencies`

Reply:
Yes, the default is to add both, and by virtue of how argparse works, there's no equivalent explicit option.  This arrangement is intended to reflect how `spack install` parses the same option.

The help string from the `install` command is noted; I used a different description since nothing is actually being "installed", but I do think it can be improved.  Feel free to suggest a help string that you think better clarifies this argument's purpose.

Reply:
Maybe "select the buildcache mode. the default is to build a cache for the package along with all its dependencies. alternatively one can decide to build a cache for only the package or only the dependencies"

Not a huge deal though. if this PR is ready to go, I wouldn't let this hold it up.